fetch.bs

```diff
@@ -2049,6 +2049,12 @@ clearly stipulates that <a>connections</a> are keyed on
 <!-- See https://github.com/whatwg/fetch/issues/114#issuecomment-143500095 for when we make
      WebSocket saner -->
 
+<h3 id=http-cache>HTTP Cache</h3>
+
+<p>A given request has an associated HTTP cache derived from the primary key(s). The primary key
+may be the top-frame origin of the document and also possibly the origin of the subframe
+associated with the request. When performing an <a>HTTP-network-or-cache fetch</a>, the associated
+HTTP cache is used to fetch the request URL.
```

Comment:
I was thinking that we'd do this much more explicit. In particular, update HTML to define and propagate the top-level origin concept appropriately and then whenever we mention HTTP cache in this document instead invoke an algorithm that returns the appropriate cache. With this approach there's a lot of action-at-a-distance which is hard to maintain and often leads to all kinds of subtle issues.

Reply:
Sounds good. Created a pull request on the html spec for defining top level origin. PTAL, thanks!
https://github.com/whatwg/html/pull/4966